c2corg_ui/static/js/yeti/yeti.js

```diff
@@ -0,0 +1,447 @@
+      'simple': 'Méthode manquante',
+      'full': 'Veuillez sélectionner une méthode pour le calcul'
+    },
+    'methode_bra': {
+      'simple': 'Méthode et BRA incompatible',
+      'full': 'La méthode MRD (débutant) est autorisée avec un BRA de 3 maximum'
+    },
+    'bra': {
+      'simple': 'BRA manquant',
+      'full': 'La valeur de BRA est manquante'
+    },
+    'altitude': {
+      'simple': 'Altitude manquante',
+      'full': 'L\'altitude est requise quand le BRA haut et bas sont différents'
+    },
+    'zoom': {
+      'simple': 'Zoom carte trop important',
+      'full': 'Veuillez zoomer au niveau ' + this.MIN_ZOOM + ' minimum'
+    },
+    'ok': 'Tout semble correct :)',
+    'yeti': 'Le service ne fonctionne pas actuellement',
+    'yeti_prefix': 'YETI Service: ',
+    'yeti_unauthorized': 'Vous devez être autorisé pour effectuer cette requête. Contactez jbl77 si vous êtes intéressé.'
+  };
+
+  /**
+   * @type {string}
+   * @export
+   */
+  this.actualError = '';
+
+  /**
+   * @type {boolean}
+   * @export
+   */
+  this.isProcessing = false;
+
+  /**
+   * @type {app.Authentication}
+   * @private
+   */
+  this.auth_ = appAuthentication;
+
+};
+
+/**
+ * Set initial scope data
+ * @private
+ */
+app.YetiController.prototype.setInitialData_ = function() {
+  // MRD
+  this.scope_['bra'] = {};
+  // MRE
+  this.scope_['rdv'] = {'none': true};
+  // MRP
+  this.setDanger_();
+  this.scope_['neigeMouillee'] = false;
+  this.scope_['tailleGroupe'] = 1;
+
+  // map
+  this.scope_['mapZoomOK'] = true;
+
+  // watch every input
+  this.scope_.$watchGroup([
+    'method',
+    'bra.haut',
+    'bra.bas',
+    'bra.altiseuil',
+    'bra.isDifferent',
+    'mapZoomOK'
+  ], (newValues, oldValues, scope) => {
+    // verif form
+    if (!this.scope_['bra']['haut']) {
+      this.formOK = false;
+      this.formOKError_ = 'bra';
+    } else if (!this.scope_['method']) {
+      this.formOK = false;
+      this.formOKError_ = 'methode';
+    } else if (
+      this.scope_['bra']['haut'] === 4 &&
+      this.scope_['method'] === 'mrd') {
+      this.formOK = false;
+      this.formOKError_ = 'methode_bra';
+    } else if (
+      this.scope_['bra']['bas'] &&
+      (this.scope_['bra']['haut'] != this.scope_['bra']['bas']) &&
+      !this.scope_['bra']['altiseuil']) {
+      this.formOK = false;
+      this.formOKError_ = 'altitude';
+    } else if (!this.scope_['mapZoomOK']) {
+      this.formOK = false;
+      this.formOKError_ = 'zoom';
+    } else {
+      this.formOK = true;
+    }
+    // set actual error
+    if (this.formOK) {
+      this.actualError = this.errors_['ok'];
+    } else {
+      this.actualError = this.errors_[this.formOKError_]['simple'];
+      if (this.formOKError_ === 'zoom') {
+        this.actualError += ' (actuel: ' + this.map_.getView().getZoom() + ')';
+      }
+    }
+    // also
+    // verif if bra = 4, method MRD forbidden
+    if (this.scope_['bra']['haut'] == 4 && this.scope_['method'] === 'mrd') {
```

Comment:
This check is done also on line 192. Maybe a single method could factorize the code and its meaning, e.g. `mrdIsNotApplicable()`?